### PR TITLE
kie-issues#1250: prepare for release branch,gpg,svn

### PR DESCRIPTION
**Pull request by jstastny-cz**
Addresses:
* apache/incubator-kie-issues#1250

* exporting images, signing and uploading to svn, also archiving in the job
* parameterize if images are deployed or not, reading from branch.yaml from apache/incubator-kie-kogito-pipelines#1206

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#21` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/21/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:635)<br/>	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$2.execute(CliGitAPIImpl.java:871)<br/>	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$GitCommandMasterToSlaveCallable.call(RemoteGitImpl.java:170)<br/>	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$GitCommandMasterToSlaveCallable.call(RemoteGitImpl.java:161)<br/>	at hudson.remoting.UserRequest.perform(UserRequest.java:211)<br/>	at hudson.remoting.UserRequest.perform(UserRequest.java:54)<br/>	at hudson.remoting.Request$2.run(Request.java:377)<br/>	at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)<br/>	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)<br/>	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)<br/>	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)<br/>	at java.base/java.lang.Thread.run(Thread.java:829)<br/>	Suppressed: hudson.remoting.Channel$CallSiteStackTrace: Remote call to builds33<br/>		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1787)<br/>		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:356)<br/>		at hudson.remoting.Channel.call(Channel.java:1003)<br/>		at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.execute(RemoteGitImpl.java:153)<br/>		at jdk.internal.reflect.GeneratedMethodAccessor826.invoke(Unknown Source)<br/>		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/>		at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/>		at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.invoke(RemoteGitImpl.java:138)<br/>		at com.sun.proxy.$Proxy205.execute(Unknown Source)<br/>		at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1222)<br/>		at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1305)<br/>		at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:129)<br/>		at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:97)<br/>		at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:84)<br/>		at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)<br/>		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)<br/>		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)<br/>		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)<br/>		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)<br/>		at java.base/java.lang.Thread.run(Thread.java:829)<br/>[Pipeline] deleteDir<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - 'https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/21/api/json?depth=0'<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: .ci/jenkins/Jenkinsfile.build-image [hunk 1]
```diff
@@ -222,6 +222,31 @@ pipeline {
...
+            when {
+                expression { return shouldExportAndGpgSign() }
+            }
+            steps {
+                script {
+                    def resultingFileName = "${getFinalImageName()}.tar.gz"
+                    def signatureFileName = "${resultingFileName}.asc"
+                    sh """
+                        docker pull ${getBuiltImageTag()}
+                        docker save ${getBuiltImageTag()} > ${resultingFileName}
```
Comment (rodrigonull) [inline .ci/jenkins/Jenkinsfile.build-image]:
```suggestion
                        docker save ${getBuiltImageTag()} | gzip > ${resultingFileName}
```

**Review by rodrigonull — CHANGES_REQUESTED** [review]

**Comment by kie-ci3** [comment]
**PR job** `#22` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/22/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:635)<br/>	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$2.execute(CliGitAPIImpl.java:871)<br/>	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$GitCommandMasterToSlaveCallable.call(RemoteGitImpl.java:170)<br/>	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$GitCommandMasterToSlaveCallable.call(RemoteGitImpl.java:161)<br/>	at hudson.remoting.UserRequest.perform(UserRequest.java:211)<br/>	at hudson.remoting.UserRequest.perform(UserRequest.java:54)<br/>	at hudson.remoting.Request$2.run(Request.java:377)<br/>	at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)<br/>	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)<br/>	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)<br/>	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)<br/>	at java.base/java.lang.Thread.run(Thread.java:829)<br/>	Suppressed: hudson.remoting.Channel$CallSiteStackTrace: Remote call to builds34<br/>		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1787)<br/>		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:356)<br/>		at hudson.remoting.Channel.call(Channel.java:1003)<br/>		at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.execute(RemoteGitImpl.java:153)<br/>		at jdk.internal.reflect.GeneratedMethodAccessor826.invoke(Unknown Source)<br/>		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/>		at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/>		at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.invoke(RemoteGitImpl.java:138)<br/>		at com.sun.proxy.$Proxy205.execute(Unknown Source)<br/>		at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1222)<br/>		at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1305)<br/>		at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:129)<br/>		at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:97)<br/>		at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:84)<br/>		at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)<br/>		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)<br/>		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)<br/>		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)<br/>		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)<br/>		at java.base/java.lang.Thread.run(Thread.java:829)<br/>[Pipeline] deleteDir<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - 'https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/22/api/json?depth=0'<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: .ci/jenkins/Jenkinsfile.build-image [hunk 2]
```diff
@@ -222,6 +222,31 @@ pipeline {
...
+                script {
+                    def resultingFileName = "${getFinalImageName()}.tar.gz"
+                    def signatureFileName = "${resultingFileName}.asc"
+                    sh """
+                        docker pull ${getBuiltImageTag()}
+                        docker save ${getBuiltImageTag()} | gzip > ${resultingFileName}
+                    """
+                    release.gpgImportKeyFromFileWithPassword(getReleaseGpgSignKeyCredsId(), getReleaseGpgSignPassphraseCredsId())
+                    release.gpgSignFileDetachedSignatureWithPassword(resultingFileName, signatureFileName, getReleaseGpgSignPassphraseCredsId())
+                    release.svnUploadFileToRepository(getReleaseSvnRepository(), getReleaseSvnCredsId(), resultingFileName, signatureFileName)
```
Comment (rodrigonull) [inline .ci/jenkins/Jenkinsfile.build-image]:
```suggestion
                    release.svnUploadFileToRepository(getReleaseSvnRepository(), getReleaseSvnCredsId(), getReleaseVersion(), resultingFileName, signatureFileName)
```
We need to also pass the release version which will be the directory where the artifacts will be published to.

**Review by porcelli — APPROVED** [review]

**Review by tiagobento — APPROVED** [review]

**Review by rodrigonull — APPROVED** [review]
LGTM

**Comment by kie-ci3** [comment]
**PR job** `#23` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/23/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
	at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)<br/>	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)<br/>	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)<br/>	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)<br/>	at java.base/java.lang.Thread.run(Thread.java:829)<br/>	Suppressed: hudson.remoting.Channel$CallSiteStackTrace: Remote call to builds29<br/>		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1787)<br/>		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:356)<br/>		at hudson.remoting.Channel.call(Channel.java:1003)<br/>		at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.execute(RemoteGitImpl.java:153)<br/>		at jdk.internal.reflect.GeneratedMethodAccessor826.invoke(Unknown Source)<br/>		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/>		at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/>		at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.invoke(RemoteGitImpl.java:138)<br/>		at com.sun.proxy.$Proxy205.execute(Unknown Source)<br/>		at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1222)<br/>		at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1305)<br/>		at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:129)<br/>		at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:97)<br/>		at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:84)<br/>		at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)<br/>		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)<br/>		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)<br/>		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)<br/>		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)<br/>		at java.base/java.lang.Thread.run(Thread.java:829)<br/>[Pipeline] deleteDir<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>Cloning repository https://github.com/jstastny-cz/incubator-kie-kogito-images.git<br/> > git init /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-images.build-and-test/incubator-kie-kogito-images/githubscm-prepare-commit-incubator-kie-kogito-images # timeout=10<br/>Fetching upstream changes from https://github.com/jstastny-cz/incubator-kie-kogito-images.git<br/> > git --version # timeout=10<br/> > git --version # 'git version 2.34.1'<br/>using GIT_ASKPASS to set credentials asf-ci token<br/> > git fetch --tags --force --progress -- https://github.com/jstastny-cz/incubator-kie-kogito-images.git +refs/heads/*:refs/remotes/origin/* # timeout=10<br/>+ wget --no-check-certificate -qO - 'https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/23/api/json?depth=0'<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by jstastny-cz** [comment]
@ricardozanini @wmedvede Since you've set yourselves as code-owners (kinda surprising without running this in mailing list), could you also be checking PRs not to block progress towards the release? Much appreciated.

**Review by ricardozanini — APPROVED** [review]
@jstastny-cz there was an approved PR from the community. You can see in the history. 

Feel free to send one adding yourself if .ci has any changes as we did with the operator.

**Comment by jstastny-cz** [comment]
@ricardozanini 2 people approving a PR is not a whole community opinion. There should have been at least HEADS-UP IMHO. Different topic though. I don't want to be a code owner unless it can be scoped to `.ci` folder, so it might leave me with pinging you guys each time I adjust ci, which in upcoming days might be frequent.

**Pull request merged**